blackhole.c

```diff
@@ -714,6 +714,8 @@ void blackhole_make_one(int index) {
 
     P[child].PI = atomic_fetch_and_add(&N_bh_slots, 1);
     P[child].Type = 5;	/* make it a black hole particle */
+    if(P[child].PI >= All.MaxPartBh)
```

Comment:
Shall we change the API for forking particles to take a particle type and make the assertion there?

Reply:
I think we agreed last round to keep fork_particle type-agnostic. If DM is being forked the check needs to be here.

Reply:
Assigning slot for the type can be made type agnostic. I'll do a PR on this PR.